meta-webserver/recipes-webadmin/cockpit/cockpit_217.bb

```diff
@@ -0,0 +1,178 @@
+SUMMARY = "Admin interface for Linux machines"
+DESCRIPTION = "Cockpit makes it easy to administer your GNU/Linux servers via a web browser"
+
+LICENSE = "LGPLv2+"
+LIC_FILES_CHKSUM = "file://COPYING;md5=4fbd65380cdd255951079008b364516c"
+
+SRC_URI += " \
+    https://github.com/cockpit-project/cockpit/releases/download/${PV}/cockpit-${PV}.tar.xz \
+    file://0001-remove-tests-dep-on-gobject-intro.patch \
+    file://0002-fix-makefile-use-copy-rule-for-unmodified-files.patch \
+    file://0003-feat-add-option-to-build-without-polkit.patch \
+    file://cockpit.pam \
+    "
+SRC_URI[md5sum] = "048fe1f3d9211786b6888e9d2d8db53c"
+SRC_URI[sha256sum] = "6bf41e2ccaebf8b2ede98937750bd33f3d9b369a1de18a1ac17170b59132136c"
+
+inherit gettext pkgconfig autotools systemd distro_features_check
+
+DEPENDS += "glib-2.0-native intltool-native gnutls"
+DEPENDS += "systemd virtual/gettext json-glib krb5 libpam"
+
+RDEPENDS_${PN} += "glib-networking"
+
+REQUIRED_DISTRO_FEATURES = "pam"
+
+COCKPIT_USER_GROUP ?= "root"
+COCKPIT_WS_USER_GROUP ?= "${COCKPIT_USER_GROUP}"
+
+EXTRA_AUTORECONF = "-I tools"
+EXTRA_OECONF = " \
+    --with-cockpit-user=${COCKPIT_USER_GROUP} \
+    --with-cockpit-group=${COCKPIT_USER_GROUP} \
+    --with-cockpit-ws-instance-user=${COCKPIT_WS_USER_GROUP} \
+    --with-cockpit-ws-instance-group=${COCKPIT_WS_USER_GROUP} \
+    --disable-doc \
+    --with-systemdunitdir=${systemd_system_unitdir} \
```

Comment:
this should pehaps be checking systemd to be available in DISTRO_FEATURES

Reply:
Maybe I don't really understand the question.
Cockpit is closely connected to systemd and the DBus API. What would be the advantage of using ${@bb.utils.filter('DISTRO_FEATURES', 'systemd', d)} ?

Reply:
systemd is not the only init system we support in OE, so if this package absolutely depends on systemd then perhaps it should ask for REQUIRED_DISTRO_FEATURE = "systemd"

see below error when not using systemd e.g.

```
14:51:00 Building world for qemuarm ...
14:51:27 Parsing recipes...WARNING: /mnt/jenkins/workspace/Yocto-world-arm/sources/meta-openembedded/meta-webserver/recipes-webadmin/cockpit/cockpit_217.bb: distro_features_check.bbclass is deprecated, please use features_check.bbclass instead
14:52:18 done.
14:52:19 WARNING: No bb files matched BBFILE_PATTERN_odroid-extras '^/mnt/jenkins/workspace/Yocto-world-arm/sources/meta-odroid/meta-odroid-extras/'
14:52:19 WARNING: No bb files matched BBFILE_PATTERN_meta-altera '^/mnt/jenkins/workspace/Yocto-world-arm/sources/meta-altera/'
14:52:23 ERROR: Nothing PROVIDES 'systemd' (but /mnt/jenkins/workspace/Yocto-world-arm/sources/meta-openembedded/meta-webserver/recipes-webadmin/cockpit/cockpit_217.bb DEPENDS on or otherwise requires it)
14:52:23 systemd was skipped: missing required distro feature 'systemd' (not in DISTRO_FEATURES)
14:52:27 ERROR: Nothing RPROVIDES 'cockpit' (but /mnt/jenkins/workspace/Yocto-world-arm/sources/meta-openembedded/meta-webserver/recipes-webadmin/cockpit/cockpit_217.bb RDEPENDS on or otherwise requires it)
14:52:27 No eligible RPROVIDERs exist for 'cockpit'
14:52:27 ERROR: Nothing RPROVIDES 'cockpit-dev' (but /mnt/jenkins/workspace/Yocto-world-arm/sources/meta-openembedded/meta-webserver/recipes-webadmin/cockpit/cockpit_217.bb RDEPENDS on or otherwise requires it)
14:52:27 No eligible RPROVIDERs exist for 'cockpit-dev'
14:52:27 ERROR: Nothing RPROVIDES 'cockpit-bridge' (but /mnt/jenkins/workspace/Yocto-world-arm/sources/meta-openembedded/meta-webserver/recipes-webadmin/cockpit/cockpit_217.bb RDEPENDS on or otherwise requires it)
14:52:27 No eligible RPROVIDERs exist for 'cockpit-bridge'

```

Reply:
Now I get it. I have now extended the "REQUIRED_DISTRO_FEATURES. Thank you.